### PR TITLE
Fix deepClone edge case

### PR DESCRIPTION
**Pull request by sunyuanSoftware**
## Description
FIX Issues #906 

## PR Type
- [x] Snippets, Tests & Tags (new snippets, updated snippets, re-tagging of snippets, added/updated tests)
- [ ] Scripts & Website & Meta (anything related to files in the [scripts folder](https://github.com/30-seconds/30-seconds-of-code/tree/master/scripts), how the repository's automated procedures work and the website)
- [ ] Glossary & Secondary Features (anything related to the glossary, such as new or updated terms or other secondary features)
- [ ] General, Typos, Misc. & Meta (everything else, typos, general stuff and meta files in the repository - e.g. the issue template)

## Guidelines
- [x] I have read the guidelines in the [CONTRIBUTING](https://github.com/30-seconds/30-seconds-of-code/blob/master/CONTRIBUTING.md) document.

### CODE REVIEW

File: snippets/deepClone.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ const deepClone = obj => {
   Object.keys(clone).forEach(
     key => (clone[key] = typeof obj[key] === 'object' ? deepClone(obj[key]) : obj[key])
   );
-  return Array.isArray(obj) ? (clone.length = obj.length) && Array.from(clone) : clone;
+  return Array.isArray(obj) ? Array.from({ length: obj.length }) : clone;
```
[the remark this answers is not in the document]
Comment (sunyuanSoftware) [reply snippets/deepClone.md]: deepClone({ a: [] }); // expect output { a: [] } not { a: 0 }

[the remark this answers is not in the document]
Comment (skatcat31) [reply snippets/deepClone.md]:
```js
deepClone({ a: [1,2,3] }) // { a: []}
```

[the remark this answers is not in the document]
Comment (Chalarangelo) [reply snippets/deepClone.md]: I think there's no really elegant way to fix the edge case, so we probably need to check if it's an array and empty to return and empty array, otherwise return the clone. I am trying to solve this in my head, but I probably need to write the code to see if there's a short way to write this. Does `Array.from(otherArray)` work or `Object.assign([],otherArray)` work? Those are the only things that come to mind apart from the spread operator.
